src/main/scala/viper/server/vsi/MessageStreamingTask.scala

```diff
@@ -76,7 +83,10 @@ abstract class MessageStreamingTask[T]() extends Callable[T] with Post {
     *
     * @param success indicates whether or not the task has ended as successfully.
     * */
-  protected def registerTaskEnd(success: Boolean): Unit = {
+  protected def registerTaskEnd(success: Boolean, logger: Logger): Unit = {
+    assert(!hasEnded)
```

Comment:
It’s worth throwing a semantically suitable exception rather than just assertion violation.